python/lsst/db/utils.py

```diff
@@ -21,49 +19,350 @@
+        if not dbExists(conn, dbName):
+            return False
+        return conn.engine.has_table(tableName, dbName)
+    elif not conn.engine.url.database:
+        return False
+    return conn.engine.has_table(tableName)
+
+
+def createTable(conn, tableName, tableSchema, dbName=None, mayExist=False):
+    """
+    Create table <tableName> in database <dbName>.
+
+    @param conn        Database connection.
+    @param tableName   Table name.
+    @param tableSchema Table schema starting with opening bracket. Note that it can
+                       NOT contain "--" or ";".
+    @param dbName      Database name.
+    @param mayExist    Flag indicating what to do if the database exists.
+
+    If database <dbName> is not set, and "use <database>" was called earlier,
+    it will use that database.
+
+    Raises TableExistsError if the table already exists and mayExist flag
+    is say to False.
+    Raises sqlalchemy exceptions.
+    """
+    if conn.engine.url.get_backend_name() == "mysql":
+        dbNameStr = "`%s`." % dbName if dbName is not None else ""
+        try:
+            conn.execute("CREATE TABLE %s`%s` %s" % \
+                        (dbNameStr, tableName, tableSchema))
+        except DBAPIError as e:
+            if e.orig.args[0] == MySqlErr.ER_NO_DB_ERROR:
+                raise InvalidDatabaseNameError(dbNameStr)
+            elif e.orig.args[0] == MySqlErr.ER_TABLE_EXISTS_ERROR:
+                if mayExist:
+                    return
+                raise TableExistsError(dbNameStr + tableName)
+            else:
+                raise
+    else:
+        raise NoSuchModuleError
+
+
+def createTableLike(conn, dbName, tableName, templDb, templTable):
+    """
+    Create table <dbName>.<tableName> like <templDb>.<templTable>
+
+    @param conn        Database connection.
+    @param dbName      Name of the database where the tables should be created
+    @param tableName   Name of the table to create
+    @param templDb     Name of the database where the template table is
+    @param templTable  Name of the template table
+
+    Raises TableExistsError if the to-be-created table already exists.
+    Raises NoSuchTableError if the template table does not exists.
+    Raises sqlalchemy exceptions.
+    """
+
+    if conn.engine.url.get_backend_name() == "mysql":
+        query = "CREATE TABLE {0}.{1} LIKE {2}.{3}".format(dbName, tableName,
+                                                           templDb, templTable)
+        try:
+            conn.execute(query)
+        except OperationalError as e:
+            if e.args[0].startswith("(OperationalError) (1050,"):
+                raise TableExistsError(dbName + '.' + tableName)
+            raise
+        except ProgrammingError as e:
+            if e.args[0].startswith("(ProgrammingError) (1146,"):
+                raise NoSuchTableError(templTable)
+            raise
+    else:
+        raise NoSuchModuleError
+
+
+def createTableFromSchema(conn, schema):
```

Comment:
We wanted to catch one specific error and that involves internal mysql error which we didn't want to leak to the client code.